services/wallet/collectibles/collectible_data_db.go

```diff
@@ -0,0 +1,164 @@
+package collectibles
+
+import (
+	"database/sql"
+	"fmt"
+	"math/big"
+
+	"github.com/status-im/status-go/services/wallet/bigint"
+	"github.com/status-im/status-go/services/wallet/thirdparty"
+)
+
+type CollectibleDataDB struct {
+	db *sql.DB
+}
+
+func NewCollectibleDataDB(sqlDb *sql.DB) *CollectibleDataDB {
+	return &CollectibleDataDB{
+		db: sqlDb,
+	}
+}
+
+const collectibleDataColumns = "chain_id, contract_address, token_id, provider, name, description, permalink, image_url, animation_url, animation_media_type, traits, background_color, token_uri"
+
+func insertOrReplaceCollectiblesData(creator statementCreator, collectibles []thirdparty.CollectibleData) error {
+	insertCollectible, err := creator.Prepare(fmt.Sprintf(`INSERT OR REPLACE INTO collectible_data_cache (%s) 
+																				VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)`, collectibleDataColumns))
+	if err != nil {
+		return err
+	}
+
+	for _, c := range collectibles {
+		_, err = insertCollectible.Exec(
+			c.ID.ContractID.ChainID,
+			c.ID.ContractID.Address,
+			(*bigint.SQLBigIntBytes)(c.ID.TokenID.Int),
+			c.Provider,
+			c.Name,
+			c.Description,
+			c.Permalink,
+			c.ImageURL,
+			c.AnimationURL,
+			c.AnimationMediaType,
+			&JSONBlob{c.Traits},
+			c.BackgroundColor,
+			c.TokenURI,
+		)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (o *CollectibleDataDB) SetData(collectibles []thirdparty.CollectibleData) (err error) {
+	var (
```

Comment:
Just a personal note that i find this not very readable with only one var. Why not the simple and concise `tx, err := o.db.Begin()`. It will still be captured given that it is the first definition.